### PR TITLE
feat: reference script modules as URL assets

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

If we use `new URL()` to reference `.js` or `.ts` files, they will be treated as URL assets and will not be processed by Rsbuild's built-in `swc-loader`.

```tsx
// foo.ts will remain the original content and be output to the dist directory
const fooTs = new URL('./foo.ts', import.meta.url).href;

console.log(fooTs); // "/static/foo.[hash].ts"
```

## Related Links

see https://github.com/web-infra-dev/rsbuild/issues/3953

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/guide/basic/static-assets.mdx [hunk 1]
```diff
@@ -37,17 +39,32 @@ Import with [alias](/guide/advanced/alias) is also available:
...
 ```tsx
 const logo = new URL('./static/logo.png', import.meta.url).href;
 
+console.log(logo); // "/static/logo.[hash].png"
+
 export default = () => <img src={logo} />;
 ```
 
+If you use `new URL()` to reference `.js` or `.ts` files, they will be treated as URL assets and will not be processed by Rsbuild's built-in `swc-loader`.
+
```
Comment (9aoy) [inline website/docs/en/guide/basic/static-assets.mdx:60]: Shall we need to add a guide on how to treat it as a URL asset and have it processed by the built-in `swc-loader` in Rsbuild?

Comment (chenjiahan) [reply website/docs/en/guide/basic/static-assets.mdx:60]: I have no idea how to support it, do you have any suggestions?

Comment (9aoy) [reply website/docs/en/guide/basic/static-assets.mdx:60]: no idea +1 😢

Comment (chenjiahan) [reply website/docs/en/guide/basic/static-assets.mdx:60]: We can find a solution and include a guide in another PR.

**Pull request merged**